### PR TITLE
fix(context module): should escape context module regex

### PR DESCRIPTION
**Pull request by suxin2017**
## Related issue (if exists)

Fixes https://github.com/web-infra-dev/rspack/issues/2960

### <samp>🤖 Generated by Copilot at da7a8be</samp>

Fix a bug in the context module of `rspack_core` that affected dynamic module resolution. Add a test fixture for issue #2960 in `rspack_plugin_javascript` to verify the fix.

<details open=true>
  <summary><h2>Walkthrough</h2></summary>

### <samp>🤖 Generated by Copilot at da7a8be</samp>

* Fix a bug in the context module that caused incorrect request matching ([link](https://github.com/web-infra-dev/rspack/pull/3037/files?diff=unified&w=0#diff-0d64b75f0221f54e5a13cc5f4b32d2ffb422b6b58415cf0444a065307d9a29a3L426-R430))
* Add a test fixture for issue #2960, which is related to the context module bug ([link](https://github.com/web-infra-dev/rspack/pull/3037/files?diff=unified&w=0#diff-57e9cb04ba6e702febc72540738ee03aa2521a70f2327ebc2375d181a61506a5L1-R45), [link](https://github.com/web-infra-dev/rspack/pull/3037/files?diff=unified&w=0#diff-c7db03713667ead687d2d838bcf7d9bd249aaba52c5aa48c8e50ee2b884b68adR1), [link](https://github.com/web-infra-dev/rspack/pull/3037/files?diff=unified&w=0#diff-5ec7b996261fbce79ed2b230d20f08c3807675feef6e7a06c14c822b850f79f5L1-L-1), [link](https://github.com/web-infra-dev/rspack/pull/3037/files?diff=unified&w=0#diff-b2ddf0170b6cc344b74091e2170ec815b78b8d874d2ea5c0a050bc9142fa1f96L1-L-1), [link](https://github.com/web-infra-dev/rspack/pull/3037/files?diff=unified&w=0#diff-df0103247e4b1bed5d5f2f5ef669c6503ddf5a500a56e2b791d78ccc77a3151aL1-L-1), [link](https://github.com/web-infra-dev/rspack/pull/3037/files?diff=unified&w=0#diff-c1707ed05fa17e5d9097889226c15e85fe92fa6aecef9b1294e12d822ce32727L1-R12))

</details>

### CODE REVIEW

**Comment by underfin** [comment]
@suxin2017 Look like it was caused because `prefix` not escapes, webpack do it at https://github.com/webpack/webpack/blob/main/lib/dependencies/ContextDependencyHelpers.js#L99. Can you aligin webpack to fix it ?

**Comment by suxin2017** [comment]
> @suxin2017 Look like it was caused because `prefix` not escapes, webpack do it at https://github.com/webpack/webpack/blob/main/lib/dependencies/ContextDependencyHelpers.js#L99. Can you aligin webpack to fix it ?

ok

**Comment by suxin2017** [comment]
> @suxin2017 Look like it was caused because `prefix` not escapes, webpack do it at https://github.com/webpack/webpack/blob/main/lib/dependencies/ContextDependencyHelpers.js#L99. Can you aligin webpack to fix it ?

I have finished the work

File: crates/rspack_plugin_javascript/src/visitors/dependency/scanner.rs [hunk 1]
```diff
@@ -504,6 +505,13 @@ fn scanner_context_module(expr: &Expr) -> Option<(String, String)> {
   }
 }
 
+#[inline]
+fn quote_meta(str: String) -> String {
+  let re =
+    Regex::new(r"[-\[\]\\/{}()*+?.^$|]").expect("Failed to initialize `metacharacters REGEXP`");
```
Comment (underfin) [inline crates/rspack_plugin_javascript/src/visitors/dependency/scanner.rs:511]: The regex instance can be static use `Lazy::new`.

**Comment by underfin** [comment]
@suxin2017 Thanks your great work 👍

File: crates/rspack_plugin_javascript/src/visitors/dependency/scanner.rs (hunk 1 above)
Comment (suxin2017) [reply crates/rspack_plugin_javascript/src/visitors/dependency/scanner.rs:511]: ok

**Review by underfin — APPROVED** [review]

**Pull request merged**